app/shop.py

```diff
@@ -0,0 +1,23 @@
+from dataclasses import dataclass
+from decimal import Decimal
+
+
+@dataclass
+class Shop:
+    name: str
+    location: list[int]
+    products: dict
+
+    def product_cost(self, product_name: str, quantity: int) -> Decimal:
+        product_cost = (Decimal(str(self.products[product_name]))
+                        * Decimal(quantity))
+        if product_cost % 1 == 0:
+            return product_cost.quantize(Decimal("0"))
+        return product_cost
+
+    def calc_total_cost(self, product_cart: dict) -> Decimal:
+        total_cost = Decimal("0")
+        for product in product_cart:
+            total_cost += (Decimal(str(self.products[product]))
+                           * Decimal(str(product_cart[product])))
+        return total_cost
```

Comment:
Are you sure these methods should be in Shop, not in Customer? Total cost will be different for each customer depending on their parameters

Reply:
It would be different to each shop as well depending on customer's list of products.